packages/vision/src/index.js

```diff
@@ -410,7 +412,9 @@ Vision.prototype.detect = function(images, options, callback) {
 
       var detections = foundImages
         .map(groupDetectionsByImage)
-        .map(removeExtraneousAnnotationObjects)
+        .map(function(image) {
+          return curation.removeExtraneousAnnotations(image, options.types);
```

Comment:
Okay, so essentially, we start with `images` but the `groupDetectionsByImage` function just the annotations. Got it.